### ISSUE TITLE
E0112: too-many-star-expressions triggers on valid code

### ISSUE THREAD

**Issue by njsmith** [PyCQA/pylint#2513]
### Steps to reproduce

Here's some legal python code:

```python
(a, *b), *c = ((1, 2, 3), 4, 5)
print(a)  # 1
print(b)  # (2, 3)
print(c)  # (4, 5)
```

For example, this [can be used](https://github.com/python-trio/trio/blob/3ae0fda74417dfc79e285012073f13fcdba1d270/trio/_socket.py#L552) to unpack the complicated return value from [socket.getaddrinfo](https://docs.python.org/3/library/socket.html#socket.getaddrinfo).

### Current behavior

Running pylint on the above code reports E0112: too-many-star-expressions:

```
/tmp/stars.py:1:0: E0112: More than one starred expression in assignment (too-many-star-expressions)
```

### Expected behavior

Don't report this error on valid code.

### pylint --version output

```
pylint 2.1.1
astroid 2.0.4
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0]
```

**Comment by PCManticore** [comment]
Thanks for the suggestion though! I'll take a look tomorrow for making that code handle more traversal levels.
